cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -0,0 +1,574 @@
+                }
+            },
+            {
+                "name": "rook-csi-rbd-node",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-rbd-node',
+                    "userKey": self.out_map['CSI_RBD_NODE_SECRET_SECRET']
+                }
+            },
+            {
+                "name": "rook-csi-rbd-provisioner",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-rbd-provisioner',
+                    "userKey": self.out_map['CSI_RBD_PROVISIONER_SECRET']
+                },
+                "pool-name": self.out_map['RBD_POOL_NAME']
+            },
+            {
+                "name": "rook-csi-cephfs-node",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-cephfs-node',
+                    "userKey": self.out_map['CSI_CEPHFS_NODE_SECRET']
+                }
+            },
+            {
+                "name": "rook-csi-cephfs-provisioner",
+                "kind": "Secret",
+                "data": {
+                    "userID": 'csi-cephfs-provisioner',
+                    "userKey": self.out_map['CSI_CEPHFS_PROVISIONER_SECRET']
+                },
+                "filesystem-name": self.out_map['CEPHFS_FS_NAME'],
+                "pool-name": self.out_map['CEPHFS_POOL_NAME']
+            },
+            {
+                "name": "rook-ceph-retain-bucket",
+                "kind": "StorageClass",
+                "data": {
+                    "endpoint": self.out_map['RGW_ENDPOINT']
+                }
+            }
+        ]
+        return json.dumps(json_out)+LINESEP
+
+    def main(self):
+        generated_output = ''
+        if self._arg_parser.format == 'json':
+            generated_output = self.gen_json_out()
+        elif self._arg_parser.format == 'bash':
+            generated_output = self.gen_shell_out()
+        else:
+            raise ExecutionFailureException("Unsupported format: {}".format(
+                self._arg_parser.format))
+        print('{}'.format(generated_output))
+        if self.output_file and generated_output:
+            fOut = open(self.output_file, 'w')
+            fOut.write(generated_output)
+            fOut.close()
+
+
+### MAIN ###
+if __name__ == '__main__':
+    rjObj = RadosJSON()
+    try:
+        rjObj.main()
+    except ExecutionFailureException as err:
+        print("Excecution Failed: {}".format(err))
+    except KeyError as kErr:
+        print("KeyError: %s", kErr)
+    except OSError as osErr:
+        print("Error while trying to output the data: {}".format(osErr))
+    finally:
+        rjObj.shutdown()
+
+
+class TestRadosJSON(unittest.TestCase):
+    def setUp(self):
+        print("{}".format("I am in setup"))
+        self.rjObj = RadosJSON(['--rbd-data-pool-name=abc',
+                                '--rgw-endpoint=localhost:9000', '--format=json'])
```

Comment:
We need a test case if the endpoint format is invalid.

Reply:
Added the testcase (as well as the argument check) for the rgw-endpoint...